### DISCUSSION TITLE
Lights Consistently Go Offline After Two Minutes

### DISCUSSION THREAD

**Discussion by Colorado4Wheeler** [make-all/tuya-local#1022] [Q&A]
Device: Lumary Smart Gimbal Recessed Light C1
Tuya-Local Auto Detected Device: RGBCW Lightbulb

I'm testing the Tuya lights + this integration on a bench so I can easily cut and apply power as needed.  When I first plug it in the light is available in HomeAssistant within about a minute of getting power, then about 2 minutes later it goes offline.  This is without any changes to its ability to talk to the Internet so it has no restrictions talking to the Tuya IOT site - however I have also tested it after blocking all access to the Internet and the results are the same.

The IOT site shows the device as online.  During the test my Smart Life app is NOT running.  When it goes offline I go into the app and can control it and as soon as I do it's available in HA again and fully controllable.  Within about 2 minutes it again goes offline.  A persistent ping to the device from the time it boots up never stops responding, even long after going offline in HA.

This sounds suspiciously similar to #1012 so I'm assuming that my lights have the same issue.  The response on that discussion seems to imply that these are BLE devices or at least BLE reliant, but I'm confused about that since these bear an IP address and are fully controllable from Tuya Local for those 2 minutes.  Is this a dual protocol thing where BLE is needed to wake them up (or keep them alive) and then TCP/IP can be used to control them?  

What's confusing to me is that the built-in Tuya integration works fine and as far as I know it's not doing any kind of BLE wakeup and doesn't time out after 2 minutes, they stay alive perpetually - however I understand that the built in integration is using Tuya's API and obviously Tuya knows how to toggle the lights.

Is it the RGBCW Lightbulb that is the issue, should I select a different device or is this just how it is and there's no way to make Tuya Local to work on devices like mine?

**Comment by make-all** [comment]
It seems some newer bulbs have both BLE and WiFi, and the app wakes them via BLE when the mobile app is used.  I'm not sure though how such lights work via the cloud. They must still be polling the cloud at least every couple of seconds to appear reasonably responsive when controlled via a voice assistant, so I'm not sure how much power savings they are really getting from this more complex setup.

**Reply by make-all** [reply]
Another possibility is that it is simply a bug in the light's firmware and they are dropping the persistent connection after 2 minutes.  If that is the case, switching to polling in the device config might help (at the cost of being slower to show external changes to the light's state in Home Assistant).
